hipcheck/src/main.rs

```diff
@@ -144,6 +145,130 @@ fn cmd_schema(args: &SchemaArgs) {
 	}
 }
 
+/// Copy individual files in dir instead of entire dir, to avoid users accidentally
+/// overwriting important dirs such as /usr/bin/
+fn copy_dir_contents<P: AsRef<Path>, Q: AsRef<Path>>(from: P, to: Q) -> Result<()> {
+	let src: PathBuf = from.as_ref().to_path_buf();
+	if !src.is_dir() {
+		return Err(hc_error!("source path must be a directory"));
+	}
+	let dst: PathBuf = to.as_ref().to_path_buf();
+	if !dst.is_dir() {
+		return Err(hc_error!("target path must be a directory"));
+	}
+
+	for entry in walkdir::WalkDir::new(&src) {
+		let src_f_path = entry?.path().to_path_buf();
+		if src_f_path == src {
+			continue;
+		}
+		let mut dst_f_path = dst.clone();
+		dst_f_path.push(
+			src_f_path
+				.file_name()
+				.ok_or(hc_error!("src dir entry without file name"))?,
+		);
+		// This is ok for now because we only copy files, no dirs
+		std::fs::copy(src_f_path, dst_f_path)?;
+	}
+
+	Ok(())
+}
+
+fn cmd_setup(config: &CliConfig) -> ExitCode {
+	// Make config dir if not exist
+	let Some(tgt_conf_path) = config.config() else {
+		print_error(&hc_error!("target config dir not specified"));
+		return ExitCode::FAILURE;
+	};
+	if !tgt_conf_path.exists() && create_dir_all(tgt_conf_path).is_err() {
+		print_error(&hc_error!("failed to create missing target config dir"));
+	}
+	let Ok(abs_conf_path) = tgt_conf_path.canonicalize() else {
+		print_error(&hc_error!("failed to canonicalize HC_CONFIG path"));
+		return ExitCode::FAILURE;
+	};
+
+	// Make data dir if not exist
+	let Some(tgt_data_path) = config.data() else {
+		print_error(&hc_error!("target data dir not specified"));
+		return ExitCode::FAILURE;
+	};
+	if !tgt_data_path.exists() && create_dir_all(tgt_data_path).is_err() {
+		print_error(&hc_error!("failed to create missing target data dir"));
+	}
+	let Ok(abs_data_path) = tgt_data_path.canonicalize() else {
+		print_error(&hc_error!("failed to canonicalize HC_DATA path"));
+		return ExitCode::FAILURE;
+	};
+
+	// Copy current `hc` binary to OS-specific bin path if not there already
+	if which::which("hc").is_err() {
```

Comment:
I think this may be holder behavior from `install.sh`. The distinction in this case is that `hc setup` would be run _after_ the `hc` binary itself is in place on the local system.